src/include/main/db_main.h

```diff
@@ -281,7 +281,31 @@ class DBMain {
           use_settings_manager_ ? BootstrapSettingsManager(common::ManagedPointer(db_main)) : DISABLED;
 
       std::unique_ptr<metrics::MetricsManager> metrics_manager = DISABLED;
-      if (use_metrics_) metrics_manager = std::make_unique<metrics::MetricsManager>();
+      if (use_metrics_) {
+        metrics_manager = std::make_unique<metrics::MetricsManager>();
+        if (use_settings_manager_) {
+          if (settings_manager->GetBool(settings::Param::metrics_logging)) {
+            auto action_context = std::make_unique<common::ActionContext>(common::action_id_t(0));
```

Comment:
What are these `action_context` for? They're not really used. Otherwise looks good!

Reply:
Whoops. Those are an artifact from fiddling with the `SettingsManager`. Removed.